sympy/polys/rootoftools.py

```diff
@@ -970,6 +970,8 @@ def _eval_Eq(self, other):
         # is_real value of the CRootOf instance.
         if type(self) == type(other):
             return sympify(self == other)
+        if other.has(Symbol):
```

Comment:
The case below would return False for `Eq(root, Integral(x, (x, 1, 10)))` but yours would return None. As was suggested, which `if` is sending back the False? It is the one making the mistake and should be fixed.

Reply:
> The case below would return False for `Eq(root, Integral(x, (x, 1, 10)))` 
but yours would return None



In my case also it is returning false -
```
In [1]: r = RootOf(x**5 - x + 1, 0)                                                                                                                                                                          

In [2]: Eq(r, Integral(x, (x, 1, 10)))                                                                                                                                                                       
Out[2]: False

```

Reply:
We need to figure out the problem before deciding how to fix it

Reply:
> In my case also it is returning false

Then it is not doing so from this function because the first check is false and your check is true and would return None.